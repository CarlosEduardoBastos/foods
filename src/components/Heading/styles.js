import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.xsmall};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xxlarge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media (max-width: 1039px) {
    font-size: ${theme.fonts.sizes.large};
  }
`;

export const Container = styled.h1`
  ${({ theme, size }) => css`
    color: ${theme.colors.secondaryColor};
    font-weight: bold;
    ${titleSize[size](theme)}
  `}
`;
