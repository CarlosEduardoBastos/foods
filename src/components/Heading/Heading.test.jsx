import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xxlarge,
    });
  });

  it('should render correct heading sizes', () => {
    renderTheme(<Heading size="small">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xsmall,
    });
  });

  it('should render with size big', () => {
    renderTheme(<Heading size="big">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xxlarge,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading size="big">texto</Heading>);
    expect(container).toMatchSnapshot();
  });
});