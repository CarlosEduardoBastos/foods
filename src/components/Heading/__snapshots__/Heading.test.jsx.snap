// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
<div>
  <h2
    class="sc-bcXHqe hnIJOY"
  >
    texto
  </h2>
</div>
`;
